cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
Yeah so the more I look at this and think about it, I think you may have been right to assume that its just the `idtoken` key that needs returned. I can't find any reason to have the `consent` or `challenge` keys in that set. I'll update to reflect that, sorry about the run around. 

On the upside we've had a chance to look at key rotation within the sets. I'll leave that for another PR though 😉